vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -1700,14 +1756,30 @@ def forward_impl(self, hidden_states: torch.Tensor,
             logical_replica_count=self.logical_replica_count,
         )
 
-        if do_naive_dispatch_combine:
-            final_hidden_states = get_ep_group().combine(final_hidden_states)
-        if self.reduce_results and (self.tp_size > 1 or self.ep_size > 1):
-            # Default set to False. (May have to add shared expert outputs.
-            final_hidden_states = self.maybe_all_reduce_tensor_model_parallel(
-                final_hidden_states)
+        if shared_output is not None:
```

Comment:
Nit: `# if the shared_output was computed, add it to the final_hidden_states`

Furthermore, I think it may be a bit safer to reconfigure this check to
```
if self.shared_experts is not None:
    assert shared_output is not None
...
```

That way we keep the check consistent throughout the function and keep ourselves safe if we are expecting a shared_output but we don't get one back from the modular kernel?

Reply:
Unfortunately, self.shared_experts always needs to be set even if we aren't doing the overlap due to weight loading/processing.  I can use `config.n_shared_experts is not None` instead.